src/components/Filters/FilterModal.tsx

```diff
@@ -1,243 +1,296 @@
-			hideSubmitButton
-		/>
-	);
-};
-
-const RelativeBox = styled(Box)`
-	position: relative;
-`;
-
-const DeleteButton = styled(Button)`
-	color: rgba(0, 0, 0, 0.4);
-	position: absolute;
-	bottom: 7px;
-	right: -35px;
-`;
-
 export const FilterModal = ({
+	schema,
+	editFilter,
 	addFilter,
 	onClose,
-	schema,
-	edit,
-	fieldCompareFn,
 }: FilterModalProps) => {
-	const [filters, setFilters] = useState(edit);
-	const [searchTerm, setSearchTerm] = useState('');
-
-	const debouncedSetSearchTerm = debounce(setSearchTerm, 300);
-
-	const setEditField = (field: string, index: number) => {
-		const currentEdit = filters.map((filter, i) =>
-			i === index ? SchemaSieve.getCleanEditModel(schema, field) : filter,
+	const properties: PropertyOption[] | undefined = React.useMemo(() => {
+		if (!schema.properties || typeof schema.properties === 'boolean') {
+			return;
+		}
+		const constructedProperties: PropertyOption[] = [];
+		Object.entries(schema.properties).forEach(
+			([key, value]: [string, JSONSchema]) => {
+				const refScheme = getPropertyScheme(value);
+				if (!refScheme?.length) {
+					constructedProperties.push({
+						key,
+						field: key,
+						title: getSchemaTitle(value, key),
+					});
+				} else {
+					refScheme.forEach((ref: string) => {
+						constructedProperties.push({
+							key: `${key}__${ref}`,
+							field: key,
+							title: getSchemaTitle(value, key, ref),
+							refScheme: ref,
+						});
+					});
+				}
+			},
 		);
-		setFilters(currentEdit);
-		setSearchTerm('');
-	};
+		return constructedProperties.sort((a, b) => (a.title > b.title ? 1 : -1));
+	}, [schema]);
 
-	const setEditOperator = (operator: string, index: number) => {
-		const currentEdit = filters.map((filter, i) =>
-			i === index ? { ...filter, operator } : filter,
-		);
-		setFilters(currentEdit);
+	const newSignature = {
+		key: !!properties?.[0].refScheme
+			? `${properties?.[0].field}__${properties?.[0].refScheme}`
+			: properties?.[0].field,
+		title: properties?.[0].title,
+		field: properties?.[0].field,
+		operator: undefined,
+		value: undefined,
+		refScheme: properties?.[0].refScheme,
 	};
 
-	const setEditValue = (value: string, index: number) => {
-		const currentEdit = filters.map((filter, i) =>
-			i === index ? { ...filter, value } : filter,
-		);
-		setFilters(currentEdit);
-	};
+	const [signatures, setSignatures] = React.useState<
+		Array<Partial<FilterSignature>>
+	>(editFilter ? getSignatures(editFilter) : [newSignature]);
 
-	const removeCompound = (index: number) => {
-		const currentEdit = filters.reduce(
-			(acc, cur, i) => (i === index ? acc : [...acc, cur]),
-			[],
-		);
-		setFilters(currentEdit);
+	const addSignature = (signature: Partial<FilterSignature>, index: number) => {
+		if (!signatures) {
+			return;
+		}
+		if (!signatures.length) {
+			setSignatures([{ ...signature }]);
+		} else if (signatures.length >= index + 1) {
+			signatures[index] = signature;
```

Comment:
Should we destructure it here, like right above?
```suggestion
			signatures[index] = {...signature};
```
Or in the other way around, do we really need the destructing in `setSignatures([{ ...signature }]);`?